### PR TITLE
Frontend change request

### PR DESCRIPTION
**Pull request by babygoat**

### CODE REVIEW

**Comment by babygoat** [comment]
Update according to discussion(https://github.com/twreporter/go-api/issues/287#issuecomment-531636471).

File: controllers/donation.go [hunk 1]
```diff
@@ -266,22 +266,25 @@ func (req clientReq) BuildTapPayReq(orderNumber, details, payMethod string) tapP
...
+		frontendRedirectUrl := "https://" + envToDonationHost[globals.Conf.Environment] + "/contribute/line/" + f + "/" + orderNumber
+
+		// Tappay server will validate the hosts provided in the result_url
+		// Wrap the backendHost to be test.twreporter.org if not in the staging or production environment
+		backendHost := ""
+		if globals.Conf.Environment == "production" || globals.Conf.Environment == "staging" {
+			backendHost = globals.Conf.App.Host
+		} else {
+			backendHost = "test.twreporter.org"
+		}
```
Comment (nickhsine) [inline controllers/donation.go]: Should we discuss the host of test environment?

In cloudflare DNS records, @taylrj already set `test-line-pay-be.twreporter.org`.

File: tests/controller_donations_test.go [hunk 2]
```diff
@@ -1101,6 +1101,158 @@ func TestLinePayNotify(t *testing.T) {
...
+		},
+		Currency:    testCurrency,
+		UserID:      user.ID,
+		OrderNumber: "ValidOrderNumber1",
+		PayMethod:   linePayMethod,
+		Status:      "fail",
+		TappayResp: models.TappayResp{
+			RecTradeID:        "ValidRecTradeID1",
+			BankTransactionID: "ValidBankTransactionID1",
+			TappayApiStatus:   null.IntFrom(421), // Gateway Timeout Error
```
[the remark this answers is not in the document]
Comment (babygoat) [reply tests/controller_donations_test.go]: Nope, `RecTradeID` and `BankTransactionID` are already obtained through the first iteration.

File: controllers/donation.go (hunk 1 above)
Comment (babygoat) [reply controllers/donation.go]: I could change it if the final testing step is determined.

Comment (babygoat) [reply controllers/donation.go]: After discussion with @taylrj , the test environment will be under the same domain `test.twrerporter.org` with the reverse proxy to determine the forwarding by path for the time being.

**Pull request merged**